### PR TITLE
[X86][MC] Allow to specify any of the 8/16/32/64 register names interchangeably for R16-R31

### PR DESCRIPTION
**Pull request by KanRobert**

### CODE REVIEW

File: llvm/lib/Target/X86/MCTargetDesc/X86MCTargetDesc.cpp [hunk 1]
```diff
@@ -817,6 +817,22 @@ MCRegister llvm::getX86SubSuperRegister(MCRegister Reg, unsigned Size,
         NO_SUB_SUPER(13, R13B)
         NO_SUB_SUPER(14, R14B)
         NO_SUB_SUPER(15, R15B)
+        NO_SUB_SUPER(16, R16B)
```
Comment (phoebewang) [inline llvm/lib/Target/X86/MCTargetDesc/X86MCTargetDesc.cpp:820]: How about define a `NO_SUB_SUPERB(8)`?

Comment (KanRobert) [reply llvm/lib/Target/X86/MCTargetDesc/X86MCTargetDesc.cpp]: Good suggestion. Done.

**Review by XinWang10 — APPROVED** [review]
LGTM

**Comment by KanRobert** [comment]
Thanks @phoebewang @XinWang10

**Pull request merged**